extensions/reddit-search/src/SubredditList.tsx

```diff
@@ -0,0 +1,163 @@
+  const queryRef = useRef<string>("");
+
+  const doSubredditSearch = async (query: string, after = "") => {
+    abortControllerRef.current?.abort();
+    abortControllerRef.current = new AbortController();
+
+    setSearching(true);
+    if (!after) {
+      setResults([]);
+    }
+
+    queryRef.current = query;
+
+    if (!query) {
+      setSearching(false);
+      return;
+    }
+
+    try {
+      const preferences = getPreferences();
+      const apiResults = await searchSubreddits(query, preferences.resultLimit, after, abortControllerRef.current);
+      for (let i = 0; i < apiResults.subreddits.length; i++) {
+        const apiResult = apiResults.subreddits[i];
+        apiResult.isFavorite = favorites.some((y) => y === apiResult.subreddit);
+      }
+
+      setSearchRedditUrl(apiResults.url);
+      if (after) {
+        setResults([...results, ...apiResults.subreddits]);
+      } else {
+        setResults(apiResults.subreddits);
+      }
+    } catch (error) {
+      if (error instanceof AbortError) {
+        return;
+      }
+
+      console.log(error);
+      showToast({
+        style: Toast.Style.Failure,
+        title: "Something went wrong :(",
+        message: String(error),
+      });
+    } finally {
+      setSearching(false);
+    }
+  };
+
+  useEffect(() => {
+    return () => {
+      abortControllerRef?.current?.abort();
+    };
+  }, []);
+
+  return (
+    <List
+      isLoading={searching}
+      onSearchTextChange={doSubredditSearch}
+      throttle
+      searchBarPlaceholder="Search Subreddits..."
+    >
+      <List.Section title="Subreddits">
+        {results.map((x) => (
+          <List.Item
+            key={x.id}
+            icon={Icon.Text}
+            title={x.title}
+            accessoryTitle={`Posted ${x.created} r/${x.subredditName}${x.isFavorite ? " ⭐" : ""}`}
+            actions={
+              <ActionPanel>
+                <Action.Push
+                  title={`Search in r/${x.subredditName}`}
+                  target={<FilterBySubredditPostList subredditName={x.subredditName} subreddit={x.subreddit} />}
+                />
+                <Action.OpenInBrowser url={x.url} icon={Icon.Globe} />
+                {!x.isFavorite && (
+                  <Action
+                    title="Add to Favorites"
+                    icon={Icon.Star}
+                    onAction={async () => {
+                      await addFavoriteSubreddit(x.subreddit);
+                      const index = results.findIndex((y) => y.id === x.id);
+                      setResults([
+                        ...results.slice(0, index),
+                        { ...results[index], isFavorite: !results[index].isFavorite },
+                        ...results.slice(index + 1),
+                      ]);
+                    }}
+                  />
+                )}
+                {x.isFavorite && (
+                  <Action
+                    title="Remove from Favorites"
```

Comment:
```suggestion
                    title="Remove From Favorites"
                    shortcut={{ modifiers: ["cmd"], key: "f" }}
```
Then it's a toggle function 🔥